src/stirling/utils/linux_headers.cc

```diff
@@ -229,7 +229,23 @@ StatusOr<std::filesystem::path> ResolvePossibleSymlinkToHostPath(const std::file
     return error::Internal(ec.message());
   }
 
-  const auto resolved_host_path = system::Config::GetInstance().ToHostPath(resolved);
+  // Relative symlinks can form an invalid path when converted to a host path when there are too
+  // many "../" references, causing the mounted host path to be lost.
+  // This happens for openSUSE's linux-header package. In order to handle
+  // this, we only use the symlink target if it is an absolute path and fall back to using the
+  // symlink name directly when it's relative path (/host/lib/modules/<uname>/build).
+  const std::filesystem::path resolved_host_path = [&resolved, &p]() -> std::filesystem::path {
```

Comment:
What is the purpose of the lambda here?

Reply:
I was trying to keep the variable `const` despite the additional branching. I'll remove it in favor of a more simple if/else check.

Reply:
Updated in 0322c23b85bcb270c6bf7259f27999fdaaff5999